src/schema/v2/artwork/collectorSignals.ts

```diff
@@ -0,0 +1,121 @@
+import { GraphQLFieldConfig } from "graphql"
+import { GraphQLInt, GraphQLObjectType } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { PartnerOfferToCollectorType } from "../partnerOfferToCollector"
+import { isFeatureFlagEnabled } from "lib/featureFlags"
+
+export const collectorSignals: GraphQLFieldConfig<any, ResolverContext> = {
+  type: new GraphQLObjectType({
+    description: "Collector signals available to the artwork",
+    name: "CollectorSignals",
+    fields: {
+      bidCount: {
+        type: GraphQLInt,
+        description: "Lot bid count",
+      },
+      lotWatcherCount: {
+        type: GraphQLInt,
+        description: "Lot watcher count",
+      },
+      partnerOffer: {
+        type: PartnerOfferToCollectorType,
+        description: "Partner offer available to collector",
+      },
+    },
+  }),
+  description: "List of collector signals",
+
+  resolve: async (artwork, {}, _ctx) => {
+    return artwork.collectorSignals
+  },
+}
+
+interface EnrichedSignals {
+  bidCount?: number
+  lotWatcherCount?: number
+  partnerOffer?: { endAt: string }
+}
+
+export const enrichArtworkWithCollectorSignals = async (artwork, ctx) => {
+  const enrichedSignals: EnrichedSignals = {}
+  artwork.collectorSignals = enrichedSignals
+
+  const artworkId = artwork._id
+
+  const inSale = artwork.sale_ids?.length > 0
+
+  const unleashContext = {
+    userId: ctx.userID,
+  }
+  const partnerOfferCollectorSignalsEnabled = isFeatureFlagEnabled(
+    "emerald_signals-partner-offers",
+    unleashContext
+  )
+  const auctionsCollectorSignalsEnabled = isFeatureFlagEnabled(
+    "emerald_signals-auction-improvements",
+    unleashContext
+  )
+
+  // Handle signals for auction artworks
+  if (inSale && auctionsCollectorSignalsEnabled) {
+    const activeSaleArtwork = await getActiveSaleArtwork(
+      {
+        artworkId,
+        saleIds: artwork.sale_ids,
+      },
+      ctx
+    )
+
+    if (activeSaleArtwork) {
+      if (artwork.recent_saves_count) {
+        enrichedSignals.lotWatcherCount = artwork.recent_saves_count
+      }
+      if (activeSaleArtwork.bidder_positions_count) {
+        enrichedSignals.bidCount = activeSaleArtwork.bidder_positions_count
```

Comment:
Can we keep these field names 1:1? Ie, if the field coming back from gravity is `recent_saves_count`, the field should be `recentSavesCount`. Ditto with `bidderPositionsCount`